internal/api/rbac/roles.go

```diff
@@ -713,41 +705,47 @@ func RoleToResources(
 	return sa, role, rb, nil
 }
 
-func replaceClaimAnnotation(sa *corev1.ServiceAccount, key string, values []string) {
-	slices.Sort(values)
-	values = slices.Compact(values)
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
+func replaceClaimAnnotation(sa *corev1.ServiceAccount, prefix string, values []rbacapi.Claim) {
+	for _, claim := range values {
+		slices.Sort(claim.Values)
+		claim.Values = slices.Compact(claim.Values)
+		if sa.Annotations == nil {
+			sa.Annotations = map[string]string{}
+		}
+		sa.Annotations[prefix+claim.Name] = strings.Join(claim.Values, ",")
 	}
-	sa.Annotations[key] = strings.Join(values, ",")
 }
 
-func amendClaimAnnotation(sa *corev1.ServiceAccount, key string, values []string) {
-	existing := sa.Annotations[key]
-	if existing != "" {
-		values = append(strings.Split(existing, ","), values...)
-	}
-	slices.Sort(values)
-	values = slices.Compact(values)
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
+func amendClaimAnnotation(sa *corev1.ServiceAccount, prefix string, values []rbacapi.Claim) {
```

Comment:
A each claim has a name and a value, so referring to an array of claims as `values` promotes a small bit of confusion. This applies to a couple other related functions as well.

```suggestion
func amendClaimAnnotation(sa *corev1.ServiceAccount, prefix string, claims []rbacapi.Claim) {
```

Reply:
values is definitely one of those words that just becomes meaningless, so absolutely makes sense. Changed.